test/react/hooks/use-room.test.tsx

```diff
@@ -0,0 +1,295 @@
+import { ChatClient, RoomLifecycle, RoomOptionsDefaults, RoomStatusListener } from '@ably/chat';
+import { act, render, renderHook } from '@testing-library/react';
+import * as Ably from 'ably';
+import React from 'react';
+import { describe, expect, it, vi } from 'vitest';
+
+import { RoomProvider, useRoom, UseRoomResponse } from '../../../src/react/index.ts';
+import { ChatClientProvider } from '../../../src/react/providers/chat-client-provider.tsx';
+import { newChatClient as newChatClientLib } from '../../helper/chat.ts';
+import { randomRoomId } from '../../helper/identifier.ts';
+
+const TestComponent: React.FC<{ callback?: (room: UseRoomResponse) => void }> = ({ callback }) => {
+  const response = useRoom();
+  if (callback) callback(response);
+  return <></>;
+};
+
+vi.mock('ably');
+
+function newChatClient() {
+  return newChatClientLib() as unknown as ChatClient;
+}
+
+describe('useRoom', () => {
+  it('it should throw an error if used outside of RoomProvider', () => {
```

Comment:
Sorry, ignore my earlier comment, we should keep all the tests descriptions as you have them below :D 
i.e
`it('should throw an error if used outside of RoomProvider'`